App info localization

dmytrolutsyk opened the pull request.

dmytrolutsyk committed:
add update category method

dmytrolutsyk committed:
update prilary category miss secondary category

dmytrolutsyk committed:
add app version and info localizations update and this commit contains to app categories update

fehmitoumi committed:
Merge branch 'DevNewApiVersion' into appInfoLocalization

fehmitoumi committed:
fixed typo

Tuni commented on appstoreconnect/api.py:
```diff
@@ -583,6 +583,45 @@ def get_build_info(self, build_id):
+		full_url = BASE_API + "/v1/apps/" + app_id + "/appInfos"
+		return self._get_resources(AppInfo, None, None, full_url)
+
+	def modify_app_info(self, app_information: AppInfo, args):
+		"""
+		:reference: https://developer.apple.com/documentation/appstoreconnectapi/modify_an_app_info
+		:return: an iterator over AppInfo resources
+		"""
+		return self._modify_resource(app_information, args)
```
Il faudrait faire comme dans les autres fonctions de modify ( cf ce que j'ai fait sur modify_app_store_version ) et mettre des arguments ( ceux dont on a besoin pour le moment ).

Idéalement faudrait tous les mettre, mais ça n'est pas ce qui est fait dans l'API aujoud'hui. Peut etre que j'ouvrirai une discussion pour demander au dev originel si on devrait pas mettre tous les params possibles ou à défaut un dict générique supplémentaire pour pallier ce manquement.

Tuni commented on appstoreconnect/api.py:
```diff
@@ -583,6 +583,45 @@ def get_build_info(self, build_id):
+		full_url = BASE_API + f"/v1/appInfos/{app_information.id}/appInfoLocalizations"
+		return self._get_resources(AppInfoLocalization, None, None, full_url)
+
+	def modify_app_store_info_localization(self, AppInfoLocalization, attributes):
+		"""
+		:reference: https://developer.apple.com/documentation/appstoreconnectapi/modify_an_app_info_localization
+		:return: an iterator over AppInfoLocalization resources
+		"""
+		return self._modify_resource(AppInfoLocalization, attributes)
+
```
même remarque pour que pour l'autre fonction modify

Tuni commented on appstoreconnect/api.py:
```diff
@@ -583,6 +583,45 @@ def get_build_info(self, build_id):
 		"""
 		return self._get_resource(Build, build_id)
 
+	def read_app_category_info(self, app_category_id):
```
les fonctions sont regroupés en dessous de commentaires # <section> , il faudrait déplacer les 3 fonctions suivantes au bon endroit; le créer si besoin

fehmitoumi committed:
added again modify_app_store_version_localization ( removed accidentaly )
added copyroth parameters form create and modify app store version functions

dmytrolutsyk committed:
Update api.py

Change localisation of functions

dmytrolutsyk committed:
change parameters of modify_app_info

dmytrolutsyk committed:
change list_app_store_info_localizations and add doc ref

dmytrolutsyk committed:
- change modify_app_store_info_localization and modify_app_store_version_localization parameters
- add reference for list_app_store_version_localizations and list_app_store_info_localizations

dmytrolutsyk committed:
Merge pull request #12 from Tuni/refactorAppInfoLocalization

Refactor app info localization
